Unified.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Sun Nov 14 12:30:49 2021
+
+@author: DianaG
+"""
+
+import ldap
+import csv
+import pandas as pd
+import datetime
+import dateutil.relativedelta
+
+# **** Part 1/2 ****
+
+ldap_server = 'ldap://172.20.1.216:389'
+base_dn = 'DC=lab, DC=visiblerisk, DC=com'
+ldap_login = 'Administrator'
+ldap_pass = 'Nhwk2iISxJuOZQa43IDch0J8z0hqB5jY'
+search_object = '(&(objectClass=user)(objectCategory=person))'
+search_attr = ['userAccountControl', 'lastLogon', 'PwdLastSet', 'memberOf']
+
+
+conn = ldap.initialize(ldap_server)
+conn.set_option(ldap.OPT_REFERRALS, 0)           
+conn.simple_bind_s(ldap_login, ldap_pass)
+result = conn.search_s(base_dn, ldap.SCOPE_SUBTREE, search_object , search_attr)
+conn.unbind()
+
+print(result[:-4])
+
+with open('/Users/DianaG/Documents/Learning/Python/AD_out.csv', 'w') as f:  
+    writer = csv.writer(f)
+    header = ['CN', 'GroupsMembership', 'UserAccountControl', 'LastLogon', 'LastPasswordSet'] 
+    writer.writerow(header)
+    for i in result[:-4]:
+        user_account_control = int(i[1].get('userAccountControl')[0])
+        last_logon = int(i[1].get('lastLogon')[0])
+        pwd_last_set = int(i[1].get('pwdLastSet')[0])
+        writer.writerow([i[0],i[1].get('memberOf'),user_account_control, last_logon, pwd_last_set])
+
+
+# **** Part 2/2 ****
+
+df = pd.read_csv('/Users/DianaG/Documents/Learning/Python/AD_out.csv')
+print(df.info() , '\n\n')
+
+print('Unique flags of UserAccountControl:' , df.UserAccountControl.unique(), ':') 
+
+### UserAccountControl flags meaning
+for i in df.UserAccountControl.unique():
+    if i == 66048:
+        print(i, ':', 'Enabled Account, Password Doesn’t Expire')
+    elif i == 66082:
+        print(i, ':', 'Disabled Account, Password Doesn’t Expire & Not Required')
+    elif i == 514:
+        print(i, ':', 'Disabled Account')
+    elif i == 512:
+        print(i, ':', 'Enabled Normal Account')
+    elif i ==546:
+        print(i, ':', 'Disabled Account, Password Not Required')
+    elif i == 2080:
+        print(i, ':', 'Interdomain Trust Account, Password Not Required', '\n\n')  
+    else:
+        raise Warning('UserAccountControl contains undefined flag!!!', i, '\n\n')
+
+### Enabled domain users  
+EnabledUsers = df.loc[df['UserAccountControl'].isin([66048, 2080, 512])] 
```

Comment:
useraccountcontrol is an integer holding several flags, one in each bit of the integer.
For example, the "disabled" flag is in the 2nd bit.
Your line should work, but it's not the correct approach, the right one should be:
df['UserAccountControl'] & 2 == 0 (the '&' is a bitwise "AND", which isolates only the 2nd bit, and ==0 checks if the bit is OFF, which means the user is not disabled, which means he is enabled)